### PR TITLE
Add callback

### PR DESCRIPTION
**Pull request by charles-cowart**
Added optional callback function parameter to a Job's qsub() and system_call() members. This allows those functions to call an external function directly when the status of a job changes to update the caller.

### CODE REVIEW

File: sequence_processing_pipeline/Job.py [hunk 1]
```diff
@@ -287,6 +306,10 @@ def qsub(self, script_path, qsub_parameters=None,
 
         if job_info['job_id']:
             # job was once in the queue
+            if callback:
```
Comment (antgonza) [inline sequence_processing_pipeline/Job.py:309]: All these should be `if callback is not None`

File: sequence_processing_pipeline/Job.py [hunk 2]
```diff
@@ -136,31 +139,25 @@ def _directory_check(self, directory_path, create=False):
...
+        :param callback: optional function taking two parameters (id, status)
+                         that is called when a running process's status is
+                         changed.
+        :return: a dictionary containing stdout, stderr, and return_code as
+                 key/value pairs.
         """
         proc = Popen(cmd, universal_newlines=True, shell=True,
                      stdout=PIPE, stderr=PIPE)
+
+        if callback:
```
Comment (antgonza) [inline sequence_processing_pipeline/Job.py:158]: `if callback is not None`

**Review by antgonza — COMMENTED** [review]
Looks fine; thank you @charles-cowart, just some minor comments.

File: sequence_processing_pipeline/Job.py (hunk 1 above)
Comment (charles-cowart) [reply sequence_processing_pipeline/Job.py]: Ty! will do!

**Comment by charles-cowart** [comment]
@antgonza Comments were resolved. This can be merged. We'll need it to resolve https://github.com/qiita-spots/qp-knight-lab-processing/issues/16

**Pull request merged**